server/api/admin.go

```diff
@@ -153,6 +153,6 @@ func (h *adminHandler) SwitchAuditMiddleware(w http.ResponseWriter, r *http.Requ
 		h.rd.JSON(w, http.StatusBadRequest, "The input is invalid.")
 		return
 	}
-	h.svr.SetAuditMiddleware(enable)
+	updatePDServerConfig(h.svr, h.svr.GetConfig(), "enable-audit", &enable)
```

Comment:
First of all, should we consider failed case caused by `h.svr.SetPDServerConfig(..)` (for example persist failed)?
And I think with using` json.Marshal` and `json.Umarshal` makes the update too conflicted. how about updating and persisting the config directly(which should be more friendly for the maintainers)?

```suggestion
	    cfg := h.svr.GetConfig().PDServerCfg
	if cfg.EnableAudit == enable {
		// TODO:
		return
	}
	cfg.EnableAudit = enable
	err = h.svr.SetPDServerConfig(cfg)
	if err != nil {
		//TODO:
		return
	}
	h.rd.JSON(w, http.StatusOK, "Switching audit middleware is successful.")
```

Reply:
I have deleted this api